### ISSUE TITLE
Allow `ToImage` to handle image paths

### ISSUE THREAD

**Issue by mantasu** [pytorch/vision#8261]
### 🚀 The feature

Was just wondering if it was possible to add support for handling image paths when using `ToImage`. Here's the desired feature:
```python
import numpy as np
from PIL import Image
from torchvision.transforms.v2 import ToImage

img_np = np.random.randint(0, 256, (100, 100, 3), dtype=np.uint8)
Image.fromarray(img_np).save("test_img.jpg")

print(ToImage()("test_img.jpg").shape) # torch.Size([3, 100, 100])
```

### Motivation, pitch

This simply allows us to avoid manually loading images, for example in custom Datsets:

```python
class MyDataset(torch.utils.data.Dataset):
    def __init__(self):
        self.transform = v2.Compose([v2.ToImage(), v2.Normalize([0.5]*3, [0.3]*3)])
    def __getitem__(self, idx):
        return self.transform(f"path/to/img-{idx}.jpg")
```

Also, would be incredibly handful when using a standalone [`to_image`](https://github.com/pytorch/vision/blob/4c0f44145792adca866a1668a79f2e11ed966491/torchvision/transforms/v2/functional/_type_conversion.py#L11) utility.

**Comment by NicolasHug** [comment]
Hi @mantasu , thanks for the feature request. Traditionally, the decoding utilities are kept separate from the transforms, as those tend to have fairly different parametrization.

Is [`read_image`](https://pytorch.org/vision/main/generated/torchvision.io.read_image.html#torchvision.io.read_image ) what you're looking for?

**Comment by mantasu** [comment]
Yeah, I was wondering if this could be added to `to_image` with an additional final check for `str`. I created a [PR](https://github.com/pytorch/vision/pull/8262) with an example functionality. But I understand if it is better to keep those separate! Feel free to close it if so~

**Comment by NicolasHug** [comment]
>  But feel free to close it if it is better to keep those separate!

Thanks for understanding @mantasu - yes, let's keep those separate. Our converstion transforms (e.g. ToImage, ToTensor, ToPILImage, etc.) have been the source of a lot of  confusion in the past, e.g. `ToTensor()` would silently scale the values of the input and convert a uint8 PIL image to float... We've tried to clean that a little, but backward-compatibility engagement make that hard. I'm afraid that adding another surface to those APIs (i.e. the ability to pass paths) would add a new layer of complexity to this already-messy space, so I prefer avoiding this. Thank you!
